### ISSUE TITLE
Unexpected arguments type error

### ISSUE THREAD

**Issue by eupraxis1981** [python/mypy#8651]
Below code provides example:

MergeStrategy = NamedTuple(
    typename="MergeStrategy",
   fields=[("left", str), ("right", str), ("how", str), ("on", List[str])],
)
Generates error
{
	"resource": <<MASKED>>,
	"owner": "python",
	"code": "error",
	"severity": 8,
	"message": "Unexpected arguments to namedtuple()",
	"source": "mypy",
	"startLineNumber": 53,
	"startColumn": 18,
	"endLineNumber": 53,
	"endColumn": 18
}

whereas this does not:
MergeStrategy = NamedTuple(
    "MergeStrategy", [("left", str), ("right", str), ("how", str), ("on", List[str])],
)

Same thing for regular namedtuple.

get same error with this:
MergeStrategy = namedtuple(typename="MergeStrategy", field_names=["left", "right", "how", "on"])

but this is fine:
MergeStrategy = namedtuple("MergeStrategy", ["left", "right", "how", "on"])

**Comment by hauntsaninja** [comment]
Shouldn't be too hard to fix, look at `parse_namedtuple_args` in semanal_namedtuple.py. Not sure if mypy's AST has the equivalent of `inspect.Signature.bind`, but that would make it really easy.

**Comment by momohatt** [comment]
I was investigating this issue and noticed that the keyword arguments `typename` and `fields` are deprecated for `typing.NamedTuple` (python/cpython#16222)

It seems that these keyword arguments are still allowed for the regular `namedtuple`, though.

**Comment by gvanrossum** [comment]
More than deprecated -- typing.NamedTuple now markes these as positional args only. But collections.namedtuple is unchanged. Not sure if there's anything actionable left here?

**Comment by momohatt** [comment]
Yes, I'm not sure if mypy should support these keyword arguments for `collections.namedtuple` when `typing.NamedTuple` doesn't accept it...
Still, it might be a good idea to elaborate the error message like: `error: mypy does not support keyword arguments 'typename' and 'field_names' for namedtuple()`
